lib/nodes/decorators/RepeatWhenNTimes.ts

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+import {ResultCodes} from '../../utils/ResultCodes';
+import {Base} from '../Base';
+import {Decorator} from './Decorator';
+
+export interface ResultConditional {
+	(result: ResultCodes): boolean;
+}
+
+export class RepeatWhenNTimes<State> extends Decorator<State> {
+
+	private conditional: ResultConditional;
+	private numRepeats: number;
+
+	constructor(desc: string, child: Base<State>, conditional: ResultConditional, numRepeats: number) {
+		super('RepeatWhenNTimes-' + desc, child);
+		this.conditional = conditional;
+		this.numRepeats = numRepeats;
+	}
+
+	onRun(state: State): Promise<ResultCodes> {
+
+		let p = this.child.onRun(state);
```

Comment:
No point in the `p` variable, just 
```
return this.child.onRun(state)
.then(...
```
should be fine

Reply:
Done